roachpb/data.go

```diff
@@ -583,9 +583,44 @@ func NewTransaction(name string, baseKey Key, userPriority UserPriority,
 	}
 }
 
+func (t *Transaction) cloneValue() Transaction {
+	// We explicitly list the fields we're cloning so that addition of a new
+	// field will cause TestTransactionClone to fail and require this method to
+	// be fixed.
+	v := Transaction{
+		Name:          t.Name,
+		Key:           t.Key,
+		ID:            t.ID,
+		Priority:      t.Priority,
+		Isolation:     t.Isolation,
+		Status:        t.Status,
+		Epoch:         t.Epoch,
+		Timestamp:     t.Timestamp,
+		OrigTimestamp: t.OrigTimestamp,
+		MaxTimestamp:  t.MaxTimestamp,
+		Writing:       t.Writing,
+		Sequence:      t.Sequence,
+	}
+	if t.LastHeartbeat != nil {
+		h := *t.LastHeartbeat
+		v.LastHeartbeat = &h
+	}
+	v.ID = append([]byte(nil), t.ID...)
+	v.CertainNodes.Nodes = append([]NodeID(nil), t.CertainNodes.Nodes...)
+	// Note that we're not cloning the span keys under the assumption that the
+	// keys themselves are not mutable.
```

Comment:
If we're willing to make this assumption we could do the same for `t.ID` (which we currently set in the composite literal and then immediately overwrite)